src/Fixer/ClassNotation/NoUnneededFinalMethodFixer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class NoUnneededFinalMethodFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'A final class must not have final methods.',
+            [
+                new CodeSample('<?php
+final class Foo {
+    final public function foo() {}
+    final protected function bar() {}
+    final private function baz() {}
+}'),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_FINAL]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $end = count($tokens) - 3; // min. number of tokens to form a class candidate to fix
+        for ($index = 0; $index < $end; ++$index) {
+            if (!$tokens[$index]->isGivenKind(T_CLASS)) {
+                continue;
+            }
+
+            $classOpen = $tokens->getNextTokenOfKind($index, ['{']);
+            $classClose = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classOpen);
+
+            $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
+            if ($prevToken->isGivenKind(T_FINAL)) {
+                $this->fixClass($tokens, $classOpen, $classClose);
```

Comment:
New logic applied.

FYI, this code was copy-pasted from [ProtectedToPrivateFixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.4.0/src/Fixer/ClassNotation/ProtectedToPrivateFixer.php), and the algorithm is a @SpacePossum optimization: https://github.com/Slamdunk/PHP-CS-Fixer/pull/1

After this, we should refactor other fixers too.